compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6844,6 +6850,115 @@ public void visit(BLangBinaryExpr binaryExpr) {
         }
     }
 
+    private BLangStatementExpression createStmtExprForNullableBinaryExpr(BLangBinaryExpr binaryExpr) {
+        BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(binaryExpr.pos);
+
+        BLangExpression tempExpr;
+
+        BUnionType exprBType = (BUnionType) binaryExpr.getBType();
+        BType nilLiftType = exprBType.getMemberTypes().iterator().next();
+        BType nullableType = BUnionType.create(null, nilLiftType, symTable.nilType);
+        ((BUnionType) nullableType).setNullable(true);
+
+        BLangSimpleVariableDef tempOne;
+
+        if (binaryExpr.lhsExpr.getBType().isNullable()) {
+            tempExpr = rewriteExpr(binaryExpr.lhsExpr);
+            tempOne = createVarDef(String.format("$temp_1"), nullableType, tempExpr, binaryExpr.pos);
+        } else {
+            tempExpr = binaryExpr.lhsExpr;
+            tempOne = createVarDef(String.format("$temp_1"), tempExpr.getBType(), tempExpr, binaryExpr.pos);
+        }
+
+        BLangSimpleVarRef tempOneRef = ASTBuilderUtil.createVariableRef(binaryExpr.pos, tempOne.var.symbol);
+        blockStmt.addStatement(tempOne);
+
+        if (tempOneRef.getBType().isNullable()) {
+            ((BUnionType) tempOne.getBType()).setNullable(false);
+        }
+
+        BLangSimpleVariableDef tempVarDef = createVarDef(String.format("result"),
+                binaryExpr.getBType(), createNilLiteral(), binaryExpr.pos);
+        BLangSimpleVarRef tempVarRef = ASTBuilderUtil.createVariableRef(binaryExpr.pos, tempVarDef.var.symbol);
+        blockStmt.addStatement(tempVarDef);
+
+        BLangSimpleVariableDef tempTwo;
+        if (binaryExpr.rhsExpr.getBType().isNullable()) {
+            tempTwo = createVarDef(String.format("$temp_2"),
+                    nullableType, binaryExpr.rhsExpr, binaryExpr.pos);
+        } else {
+            tempTwo = createVarDef(String.format("$temp_2"),
+                    binaryExpr.rhsExpr.getBType(), binaryExpr.rhsExpr, binaryExpr.pos);
+        }
```

Comment:
Since this change nullable to false we cannot use the same variable, because when that variable referred somewhere again it will not be consider as a nullable.  Because of that it is necessary declare new variables instead of using the same